making-aeternity-wallet-transactions-with-aepp-sdk-js.md

```diff
@@ -0,0 +1,134 @@
+        <script src="https://bundle.run/buffer@5.6.0"></script>
+    </head>
+</html>
+```
+Lastly, create an index.js file in the same directory and include it in your index.html file just after the links to the buffer.This is the file where we will be doing most of our work in, your index.html file should look like this after
+```html
+full head of the index.html file
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <title>Wallet Web Tutorial</title>
+        <script src="https://unpkg.com/@aeternity/aepp-sdk@7.3.1/dist/aepp-sdk.browser-script.js"></script>
+        <script src="https://bundle.run/buffer@5.6.0"></script>
+        <script src="./index.js"></script>
+    </head>
+    <body>
+        
+    </body>
+</html>
+```
+With the above done, we are set to start doing wonders with the aeternity javascript SDK
+ ## Wallet creation
+ Creating an Aeternity wallet with the javascript SDK is extremely simple since we only need one module from the SDK to do this, which is the `Crypto` module, to have access to a wallet, we need to create a KeyPair(An object containing a private key and a public key). We will go over two ways to create this and they are:
+ - KeyPair creation with an existing private key
+ - New KeyPair creation without a private key
+ 
+ ### KeyPair Creation with an existing private key
+ Lets see how this is done, write the following code in the `index.js` file created earlier:
+ ```javascript
+ function generateKeyPairWithExistingPrivateKey(secretKey){
+    const hexBuffer = Ae.Crypto.hexStringToByte(secretKey);   
+      const keyPair =  Ae.Crypto.generateKeyPairFromSecret(hexBuffer)   
+      const publicKey=Ae.Crypto.aeEncodeKey(keyPair.publicKey);
+      const secretKey=buffer.Buffer.from(keyPair.secretKey).toString('hex')
+      return{publicKey:publicKey,privateKey:secretKey};
+    }
+ ```
+ What we do above is pretty simple, the function above expects a private key as a string and then converts it to an ArrayBuffer using the `hexStringToByte` function in the crypto module of the aepp-sdk-js. We further access the `generateKeyPairFromSecret` function in the crypto module of the aeternity javascript SDK, this function then returns a KeyPair object containing both the secret key and private key as an ArrayBuffer of view UInt8Array. The next step is to retrieve the private key and public key from their respective ArrayBuffers, we do this using the `aencodeKey` function of the `Crypto` module and the `from` method of the Buffer module to get the private key and public Key as strings. So now that you know how to get a public key from an existing private key. let's go over creating a KeyPair from a brand new public key and private key.
+ 
+ ### New KeyPair Creation Without Existing Public Key
+```javascript
+ function generateKeyPairWithoutSecretKey(){
```

Comment:
I think its better to connect to existing wallet, keypairs shouldn't be dealt with manually

Reply:
Yeah, there is  a function that i created and explained in the tutorial that also shows how to do that

Reply:
So have deleted the part that explains how to do this and left the part that explains how to connect to an existing wallet using its private key